Source/ACE/Entity/Scroll.cs

```diff
@@ -0,0 +1,157 @@
+﻿// WeenieType.Scroll
+
+using ACE.DatLoader.FileTypes;
+using ACE.Entity.Actions;
+using ACE.Entity.Enum;
+using ACE.Entity.Enum.Properties;
+using ACE.Network;
+using ACE.Network.GameEvent.Events;
+using ACE.Network.GameMessages.Messages;
+using ACE.Network.Motion;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace ACE.Entity
+{
+    public class Scroll : WorldObject
+    {
+        private static readonly UniversalMotion motionReading = new UniversalMotion(MotionStance.Standing, new MotionItem(MotionCommand.Reading));
+        private static readonly UniversalMotion motionReady = new UniversalMotion(MotionStance.Standing, new MotionItem(MotionCommand.Ready));
+
+        public Scroll(AceObject aceObject)
+            : base(aceObject)
+        {
+            SpellTable table = SpellTable.ReadFromDat();
+
+            Use = $"Inscribed spell: {table.Spells[SpellId].Name}\n";
+            Use += $"{table.Spells[SpellId].Desc}";
+
+            LongDesc = "Use this item to attempt to learn its spell.";
+
+            Power = table.Spells[SpellId].Power;
+            School = table.Spells[SpellId].School;
+        }
+
+        private uint SpellId
+        {
+            get { return (uint)Spell.Value; }
+        }
+
+        private uint Power
+        {
+            get;
+            set;
+        }
+
+        private MagicSchool School
+        {
+            get;
+            set;
+        }
+
+        public override void OnUse(Session session)
+        {
+            bool success = true;
+            string failReason = "You are unable to read the scroll.";
+
+            switch (Power)
+            {
+                // research: http://asheron.wikia.com/wiki/Announcements_-_2002/06_-_Castling
+                case 50:  // Level 2
+                case 100: // Level 3
+                case 150: // Level 4
+                case 200: // Level 5
+                case 250: // Level 6
+                    if (session.Player.CheckMagicSkillLevel(School, Power))
+                        success = true;
+                    else
+                    {
+                        success = false;
+                        failReason = "You are not skilled enough in the inscribed spell's school of magic to understand the writing on this scroll.";
+                    }
+                    break;
+                default: // Level 1 or Level 7+ never fail
+                    success = true;
+                    break;
+            }
+
+            if (!session.Player.UnknownSpell(SpellId))
+            {
+                success = false;
+                failReason = "You already know the spell inscribed upon this scroll.";
```

Comment:
not how I recall it. If you knew the spell, you still "read" the scroll then get the "you already know this error"

I don't have screens of the actual fail messages, so I'm just going off of memory. could use a pcap or two as well of failures but, needle in a haystack